### PR TITLE
Fixes some route creation and updating bugs.

### LINKED ISSUE #539: error: routes creation takes at least three arguments: app name, route path and image name
When I execute `fn routes create myapp /hello` following the steps and example from the README file it fails with the message: *error: routes creation takes at least three arguments: app name, route path and image name*

## Docker version:
```sh
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 08:47:51 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 08:47:51 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

## Workaround
- Execute `docker ps` check for the image name and *TAG* and run `fn routes create myapp /hello $USER/hello:$TAG`. For example `fn routes create myapp /hello nesterran/hello:0.0.1`

### PR DESCRIPTION
**Pull request by treeder**
Fixes #539

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,14 +129,17 @@ and deploy it.
...
 # build the function
 fn build
-# test it
+# test it - you can pass data into it too by piping it in, eg: `cat hello.payload.json | fn run`
 fn run
-# push it to Docker Hub
-fn push
-# create an app
+# Once it's ready, bump, build and push it to Docker Hub
+fn bump && fn build && fn push
```
Comment (jconning) [inline README.md]: `fn bump` is actually not required when a new function is being built for the first time.  This section of the quickstart is only covering the case of a new function, not editing an existing function.  The way this is written the first version of the function will be 0.0.2.

`fn bump` is required when the code of an existing function is edited.

File: README.md [hunk 2]
```diff
@@ -129,14 +129,17 @@ and deploy it.
...
+# create an app - you only do this once per app
 fn apps create myapp
 # create a route that maps /hello to your new function
 fn routes create myapp /hello
+# to update a function with a new version run
+fn bump && fn build && fn push
+fn routes update myapp /hello
```
Comment (jconning) [inline README.md]: Looks good.

**Comment by treeder** [comment]
Updated.

File: fn/routes.go [hunk 3]
```diff
@@ -278,7 +278,8 @@ func envAsHeader(req *http.Request, selectedEnv []string) {
 }
 
 func (a *routesCmd) create(c *cli.Context) error {
-	if c.Args().Get(0) == "" || c.Args().Get(1) == "" || c.Args().Get(2) == "" {
+	// todo: @pedro , why aren't you just checking the length here?
+	if c.Args().Get(0) == "" || c.Args().Get(1) == "" {
```
Comment (pedronasser) [inline fn/routes.go]: @treeder I haven't changed so many things here. @ccirello checked like that, and I just increase one more argument validation.
But just validate the length is a better solution.

File: README.md [hunk 4]
```diff
@@ -129,11 +129,11 @@ and deploy it.
 fn init $USERNAME/hello
 # build the function
 fn build
-# test it
+# test it - you can pass data into it too by piping it in, eg: `cat hello.payload.json | fn run`
 fn run
-# push it to Docker Hub
-fn push
-# create an app
+# Once it's ready, bump, build and push it to Docker Hub
```
Comment (jconning) [inline README.md]: "bump, build and push" should actually be "build and push".

File: fn/routes.go [hunk 5]
```diff
@@ -278,7 +278,8 @@ func envAsHeader(req *http.Request, selectedEnv []string) {
...
 		return errors.New("error: routes creation takes at least three arguments: app name, route path and image name")
```
Comment (jconning) [inline fn/routes.go]: Error message doesn't match logic.  Logic is checking app name and route path only.  But error message also mentions image name as required.

File: fn/routes.go (hunk 3 above)
Comment (treeder) [inline fn/routes.go]: Fixed.

File: fn/routes.go (hunk 5 above)
Comment (treeder) [inline fn/routes.go]: Fixed.

File: README.md (hunk 4 above)
Comment (treeder) [inline README.md]: Fixed.

File: fn/routes.go [hunk 6]
```diff
@@ -454,18 +456,48 @@ func (a *routesCmd) patchRoute(appName, routePath string, r *fnmodels.Route) err
...
-
+	image := c.Args().Get(2)
 	var (
 		format  string
 		maxC    int
 		timeout time.Duration
 	)
+	// if image == "" {
+	// todo: why do we only load the func file if image isn't set?  Don't we need to read the rest of these things regardless?
+	ff, err := loadFuncfile()
```
Comment (Gouthamve) [inline fn/routes.go]: Does this mean I cannot update an app without the funcfile?

File: fn/routes.go [hunk 7]
```diff
@@ -485,7 +517,7 @@ func (a *routesCmd) update(c *cli.Context) error {
 
 	to := int64(timeout.Seconds())
 	patchRoute := &fnmodels.Route{
-		Image:          c.String("image"),
+		Image:          image,
 		Memory:         c.Int64("memory"),
```
Comment (Gouthamve) [inline fn/routes.go]: Shouldn't these also come from the funcfile?

Comment (Gouthamve) [inline fn/routes.go]: These as in Memory/Time.

File: fn/routes.go [hunk 8]
```diff
@@ -454,18 +456,48 @@ func (a *routesCmd) patchRoute(appName, routePath string, r *fnmodels.Route) err
...
-	if c.Args().Get(0) == "" || c.Args().Get(1) == "" {
-		return errors.New("error: route configuration description takes two arguments: an app name and a route")
+	if len(c.Args()) < 2 {
+		return errors.New("error: route update takes at least two arguments: an app name and a path")
 	}
 
 	appName := c.Args().Get(0)
 	route := c.Args().Get(1)
-
+	image := c.Args().Get(2)
```
Comment (Gouthamve) [inline fn/routes.go]: Need to pick up the image from the flag. The third argument is undocumented.

Comment (treeder) [inline fn/routes.go]: It's documented in the main help. Optional now.

**Pull request merged**